### ISSUE TITLE
Add citation information to return, including retrieved date

### ISSUE THREAD

**Issue by dschristianson** [BASIN-3D/basin3d#97]
**Is your feature request related to a problem? Please describe.**
Citation information should be included in the return, including the date of retrieval.

**Describe the solution you'd like**
OGC has data model for this. Need to look into it.
Double check requirements with Charu. From planning doc: "Adding a data citation in the returned metadata query/Add versioning to metadata query (date when the data were pulled. Version as the date in the citation - Retrieved on.. )"

**Comment by vchendrix** [comment]
We need to consider capture the datasource resources used (e.g endpoints and queries, databases, filesystems.)
Probably need put the plugin version (which we don't have).
BASIN-3D version.

**Comment by vchendrix** [comment]
MVP
+ datasource query and date it was queried.

**Comment by dschristianson** [comment]
NOTE: If we use an aggregator data source do we want to preserve the provenance. E.g., CUASHI that handles various other data sources.

**Comment by charuleka** [comment]
Yes, we would want to preserve the CUAHSI aggregator info as well.
